### ISSUE TITLE
`module-foundry` to compile native modules

### ISSUE THREAD

**Issue by gejgalis** [rlidwka/sinopia#152]
Regarding to https://github.com/joyent/node/issues/8005 and other problems related to compile native modules on Windows machines I have come up into idea to support `module-foundry` in sinopia

More about `module-foundry` http://blog.nodejitsu.com/npm-binary-modules-and-module-foundry/

Let's imagine following sinopia configuration:

```
module-foundry:
    enabled: true
    win32: http://win32-module-foundry-server.my-company.org:1337
    win64: http://win64-module-foundry-server.my-company.org:1337
    linux32 http://linux32-module-foundry-server.my-company.org:1337
    mac86 http://mac32-module-foundry-server.my-company.org:1337
```

And based on `"user-agent":"npm/1.4.28 node/v0.10.32 win32 x64"` sinopia could recognize the client and in case of native module proper module-foundry server could be used to send back compiled.tgz

What do you think guys about such improvement?

**Comment by justrhysism** [comment]
Yeah I have many issues trying to get Sinopia to install correctly on our repo server - I don't want to have to install build tools on that server.
